### PR TITLE
Update MQTT update platform documentation

### PR DESCRIPTION
**Pull request by bieniu**
## Proposed change

Update of the MQTT `update` platform documentation. Adds documentation for processing `JSON` state updates via `state_topic`.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/81131
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/update.mqtt.markdown [hunk 1]
```diff
@@ -165,21 +169,33 @@ qos:
   required: false
   type: integer
   default: 0
+release_summary:
+  description: Summary of the release notes or changelog. This is not suitable for long changelogs but merely suitable for a short excerpt update description of max 255 characters.
```
Comment (jbouwh) [inline source/_integrations/update.mqtt.markdown]:
```suggestion
  description: Summary of the release notes or changelog. This is suitable a brief update description of max 255 characters.
```

File: source/_integrations/update.mqtt.markdown [hunk 2]
```diff
@@ -165,21 +169,33 @@ qos:
...
+  type: string
 retain:
   description: If the published message should have the retain flag on or not.
   required: false
   type: boolean
   default: false
 state_topic:
-  description: The MQTT topic subscribed to receive an update of the installed version.
-  required: true
+  description: The MQTT topic subscribed to receive state updates. The state update may be either JSON or a simple string with `installed_version` value. When a JSON payload is detected, the state value of the JSON payload could supply the `installed_version`, `latest_version`, `title`, `release_summary`, `release_url` or `entity_picture` values.
```
Comment (jbouwh) [inline source/_integrations/update.mqtt.markdown]:
```suggestion
  description: The MQTT topic subscribed to receive state updates. The state update may be either JSON or a simple string with `installed_version` value. When a JSON payload is detected, the state value of the JSON payload should supply the `installed_version` and can optional supply:  `latest_version`, `title`, `release_summary`, `release_url` or an `entity_picture` URL.
```

File: source/_integrations/update.mqtt.markdown [hunk 3]
```diff
@@ -165,21 +169,33 @@ qos:
...
+  required: false
+  type: string
+title:
+  description: Title of the software. This helps to differentiate between the device or entity name versus the title of the software installed.
```
Comment (jbouwh) [inline source/_integrations/update.mqtt.markdown]:
```suggestion
  description: Title of the software, or firmware update. This helps to differentiate between the device or entity name versus the title of the software installed.
```

File: source/_integrations/update.mqtt.markdown [hunk 4]
```diff
@@ -165,21 +169,33 @@ qos:
...
+  required: false
   type: string
 unique_id:
   description: An ID that uniquely identifies this Select. If two Selects have the same unique ID Home Assistant will raise an exception.
   required: false
   type: string
 value_template:
-  description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract the installed version value."
+  description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract the installed version value. Alternatively `value_template` can be used to render to a valid JSON payload."
```
Comment (jbouwh) [inline source/_integrations/update.mqtt.markdown]:
```suggestion
  description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract the `installed_version` state value or to render to a valid JSON payload on from the payload received on `state_topic`."
```

**Review by jbouwh — COMMENTED** [review]
A few comments

**Review by jbouwh — CHANGES_REQUESTED** [review]
`latest_version_topic` is not required anymore.

**Review by jbouwh — DISMISSED** [review]
Looks good to me!

**Comment by jbouwh** [comment]
@bieniu You need to fix the errors before we can merge

**Comment by bieniu** [comment]
Any hint what's wrong with line 162?

**Review by jbouwh — DISMISSED** [review]
Looks good to me!
Thnx @bieniu !

**Review by frenck — APPROVED** [review]
Thanks, @bieniu 👍

**Pull request merged**